### PR TITLE
[cli-trel] add CLI command to enable/disable TREL

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit adds commands to enable or disable TREL radio link:

```
> trel enable
Done
> trel diable
Done
```

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -4544,6 +4547,24 @@ otError Interpreter::ProcessMacSend(Arg aArgs[])
...
+    otError error = OT_ERROR_INVALID_COMMAND;
+
+    if (aArgs[0] == "enable")
+    {
+        error = otPlatTrelUdp6SetTestMode(mInstance, true);
+    }
+    else if (aArgs[0] == "disable")
+    {
+        error = otPlatTrelUdp6SetTestMode(mInstance, false);
+    }
```
Comment (librasungirl) [inline src/cli/cli.cpp:4562]: use `ParseEnableOrDisable ` instread?

https://github.com/openthread/openthread/blob/deb2a4c97a95242ed858ddef5d79e00271f7d798/src/cli/cli.cpp#L255-L273

File: src/posix/platform/trel_udp6.cpp [hunk 2]
```diff
@@ -620,11 +624,37 @@ otError otPlatTrelUdp6SendTo(otInstance *        aInstance,
...
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aEnable != sEnabled);
+
+    if (aEnable)
+    {
+        VerifyOrExit(sInitialized, error = OT_ERROR_FAILED);
```
Comment (librasungirl) [inline src/posix/platform/trel_udp6.cpp:637]: nit: New error code `OT_ERROR_FAILED` should be documented in header
https://github.com/openthread/openthread/blob/b7751460e04a233589393fd4faccef5bc6e8e88b/include/openthread/platform/trel-udp6.h#L139-L143

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -4544,6 +4547,24 @@ otError Interpreter::ProcessMacSend(Arg aArgs[])
 }
 #endif
 
+#if OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
+otError Interpreter::ProcessTrel(Arg aArgs[])
+{
+    otError error = OT_ERROR_INVALID_COMMAND;
+
+    if (aArgs[0] == "enable")
```
Comment (librasungirl) [inline src/cli/cli.cpp:4555]: may show the state of trel when no arg?

Comment (simonlingoogle) [reply src/cli/cli.cpp:4555]: not possible because there is no platform API to get the `TestMode` state.

Comment (librasungirl) [reply src/cli/cli.cpp:4555]: Not sure if there is scenario that tester need to check trel state, if there is, may consider adding one API?

**Review by abtink — APPROVED** [review]
LGTM. 👍

Just want to echo @librasungirl  suggestions on use of `ParseEnableOrDisable()` and documenting the new error code. Looks good. Thanks.

File: src/cli/cli.cpp (hunk 3 above)
Comment (simonlingoogle) [reply src/cli/cli.cpp:4555]: I think it's a good idea to allow checking TREL state, but we can do it later if it turn out to be necessary.

For now, the TREL state is determinate: 
- If we start `otbr-agent` with a TREL URL, then TREL is `enabled` by default, and can be disabled and enabled later.
- If we start `otbr-agnet` without a TREL URL, then TREL is `disabled` by default and can not be enabled. 

So, the TREL state is actually known beforehand without a checking.

Comment (librasungirl) [reply src/cli/cli.cpp:4555]: Fine to leave it to future when required.

File: src/cli/cli.cpp (hunk 1 above)
Comment (simonlingoogle) [reply src/cli/cli.cpp:4562]: done

File: src/posix/platform/trel_udp6.cpp (hunk 2 above)
Comment (simonlingoogle) [reply src/posix/platform/trel_udp6.cpp:637]: done

**Review by librasungirl — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**